homeassistant/components/sensor/qbittorrent.py

```diff
@@ -0,0 +1,140 @@
+"""Support for monitoring the qBittorrent API."""
+import logging
+
+import voluptuous as vol
+
+from requests.exceptions import RequestException
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, CONF_PASSWORD, CONF_URL, CONF_USERNAME, STATE_IDLE)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.exceptions import PlatformNotReady
+
+REQUIREMENTS = ['python-qbittorrent==0.3.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPE_CURRENT_STATUS = 'current_status'
+SENSOR_TYPE_DOWNLOAD_SPEED = 'download_speed'
+SENSOR_TYPE_UPLOAD_SPEED = 'upload_speed'
+
+DEFAULT_NAME = 'qBittorrent'
+
+SENSOR_TYPES = {
+    SENSOR_TYPE_CURRENT_STATUS: ['Status', None],
+    SENSOR_TYPE_DOWNLOAD_SPEED: ['Down Speed', 'kB/s'],
+    SENSOR_TYPE_UPLOAD_SPEED: ['Up Speed', 'kB/s'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.url,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+async def async_setup_platform(hass, config, add_entities,
+                               discovery_info=None):
+    """Set up the qbittorrent sensors."""
+    from qbittorrent.client import Client, LoginRequired
+
+    try:
+        client = Client(config.get(CONF_URL))
```

Comment:
Use `dict[key]` for required config keys and config keys with default config schema values.